### ISSUE TITLE
[SF][Table]: table header row's role is not accurate

### ISSUE THREAD

**Issue by summer2200** [SAP/ui5-webcomponents-react#6774]
### Describe the bug

An a11y issue has been reported which says the table's header is not announced as header row when focused.
Currently it has `role="row"` but seems not accurate.
![Image](https://github.com/user-attachments/assets/99ea8c83-aadd-4e3b-be32-4b89db499f6f)

### Reproduction steps

1.
2.
3.
...

### UI5 Web Components for React Version

2.5.0

### UI5 Web Components Version

2.5.0

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Hi @summer2200 

please follow our issue template. The issue lacks clear details about the expected behavior.  I've checked and `row` is the correct role for a table row and each cell of that row also has the role `columnheader`.

**Comment by summer2200** [comment]
Hi @Lukas742 , i've added expected behavior and also a video for understanding the issue. Please let me know if more details are needed. Thanks!

**Comment by Lukas742** [comment]
@summer2200 thank you :) 

I'll forward the issue to the ui5-webcomponents repo as the `Table` is developed there.

___

Hi Colleagues,
please take over.
